### PR TITLE
feat: customize dictionary's fullindex option with metadata.toml

### LINKED ISSUE #995: feature: use metadata.toml to control the creation of fullindex of single dictionary
**Is your feature request related to a problem? Please describe.**
There are cases when users do not want to fullindex the dictionary to save the disk space.

**Describe the solution you'd like**

use metadata.toml and a new add field called `fullindex`,if no specified , the value default to `true` ,  this field value should be combined with goldendict's fullindex setting.

fullindex && enabled_fullindex(in the preference)

```toml
categories = [ "English", "Russian", "Chinese" ]
fullindex=false   # or 0/1 or on/off

# the following fields have not supported yet.
[metadata]
name = "New Name"
langfrom = "English"
langto = "Russian"
```

**Describe alternatives you've considered**

create a specila folder for all the dictionaries which do no need to create the fullindex .
The folder name can be ,for example ,   `nofts`.

Any dictionaries in this very folder will not  create the fullindex.

**Additional context**

### PR DESCRIPTION
**Pull request by xiaoyifang**
fix #995 

image

image

### CODE REVIEW

File: website/mkdocs.yml [hunk 1]
```diff
@@ -42,7 +42,7 @@ nav:
       - Custom Stylesheet & JavaScript: topic_userstyle.md
       - Portable Mode: topic_portablemode.md
       - Custom transliteration: topic_transliteration.md
-      - Customize Dictionary Name: custom_dictionary_name.md
+      - Customize Dictionary Name: custom_dictionary.md
```
Comment (shenlebantongying) [inline website/mkdocs.yml:45]:
```suggestion
      - Customize Dictionary: custom_dictionary.md
```

The title is the first part, the second part will decide url,

**Comment by shenlebantongying** [comment]
The "FullIndex" is too weird. It should be "Full-text search" (fts)

Nobody uses the word "fullindex" https://www.google.com/search?q="fullindex"

**Review by shenlebantongying — CHANGES_REQUESTED** [review]

File: website/docs/custom_dictionary.md [hunk 2]
```diff
@@ -32,3 +38,24 @@ For example,
 
 ```
 
+## Disable the fullindex of the dictionary
+
+```toml
+fullindex=false
```
Comment (shenlebantongying) [inline website/docs/custom_dictionary.md:44]:
```suggestion
fts=false
```

File: website/docs/custom_dictionary.md [hunk 3]
```diff
@@ -32,3 +38,24 @@ For example,
...
+
+[metadata]
+name="New Name"
+
+```
+
+this `fullindex` field's value can be on/off, 1/0 ,true/false etc.
```
Comment (shenlebantongying) [inline website/docs/custom_dictionary.md:51]:
```suggestion
this `fts` field's value can be on/off, 1/0 ,true/false etc.
```

File: website/docs/custom_dictionary.md [hunk 4]
```diff
@@ -32,3 +38,24 @@ For example,
...
+
+```
+fullindex=false
```
Comment (shenlebantongying) [inline website/docs/custom_dictionary.md:54]:
```suggestion
fts=false
```

File: website/docs/custom_dictionary.md [hunk 5]
```diff
@@ -32,3 +38,24 @@ For example,
 
 ```
 
+## Disable the fullindex of the dictionary
```
Comment (shenlebantongying) [inline website/docs/custom_dictionary.md:41]:
```suggestion
## Disable full-text search for certain dictionary
```

File: website/docs/custom_dictionary.md [hunk 6]
```diff
@@ -32,3 +38,24 @@ For example,
...
+name="New Name"
+
+```
+
+this `fullindex` field's value can be on/off, 1/0 ,true/false etc.
+
+```
+fullindex=false
+```
+will disable the current dictionary's fullindex feature.
```
Comment (shenlebantongying) [inline website/docs/custom_dictionary.md:56]:
```suggestion
will disable the current dictionary's full-text search.
```

File: website/docs/custom_dictionary.md [hunk 7]
```diff
@@ -32,3 +38,24 @@ For example,
...
+
+
+you can check the fullindex feature on each dictionary info dialog.
```
Comment (shenlebantongying) [inline website/docs/custom_dictionary.md:59]:
```suggestion
you can check the full-text search on each dictionary's info dialog.
```

**Review by shenlebantongying — CHANGES_REQUESTED** [review]

File: src/ui/dictinfo.cc [hunk 8]
```diff
@@ -20,6 +20,8 @@ void DictInfo::showInfo( sptr< Dictionary::Class > dict )
   setWindowTitle( QString::fromUtf8( dict->getName().data(), dict->getName().size() ) );
 
   ui.dictionaryId->setText( QString::fromStdString( dict->getId() ) );
+  ui.enableFullindex->setPixmap( dict->canFTS() ? QPixmap( ":/icons/fts_enabled.svg" ) :
+                                                  QPixmap( ":/icons/fts_disabled.svg" ) );
```
[the remark this answers is not in the document]
Comment (xiaoyifang) [reply src/ui/dictinfo.cc]: image seems more eye-catching.but text can do .

**Pull request merged**